### DISCUSSION TITLE
Implementing a Strategy Events Collector for Rust Live Mode

### DISCUSSION THREAD

**Discussion by DeevsDeevs** [nkaz001/hftbacktest#166] [Q&A]
Hi there!

First off, thank you for this amazing project—I've been exploring it and learning a lot. Right now, I'm experimenting with live trading on Bybit using Rust, and I have an idea to write a simple events collector for my strategy to store data in Clickhouse.

The collector wouldn't just handle events from the exchange but also other messages reported by the strategy. I'm envisioning it as a separate actor (like a connector) that receives messages via IPC and then handles the saving logic.

I would love to hear your thoughts or advice on implementing something like this. Are there alternative approaches you'd recommend for this use case?

**Comment by nkaz001** [comment]
I don't see anything wrong with it. But, why not separate the two processes: receiving events from the exchange and receiving events from the strategy? In my opinion, the simpler the working unit, the better.

**Reply by DeevsDeevs** [reply]
I feel duplicating the same functionality isn't ideal—it’s better to have a single DB connector that can process different types of events in different ways. Additionally, if actor instances (e.g., live strategy bots) could interchange messages of any type/tag, it could support both logging and inter-strategy communication.

Do you think this fits with the framework's intended architecture, or would it introduce unnecessary complexity?

**Reply by nkaz001** [reply]
I think it totally depends on your use case. If you need an IPC mechanism between your components, that's fine. But, if it's just for logging data, I would prefer using a log collection framework with a flat log file
